sdk/go/common/workspace/project.go

```diff
@@ -358,16 +358,16 @@ func ValidateProject(raw interface{}) error {
 	var errs *multierror.Error
 	var appendError func(err *jsonschema.ValidationError)
 	appendError = func(err *jsonschema.ValidationError) {
-		if err.InstanceLocation != "" && err.Message != "" {
+		if err.InstanceLocation != "" && err.Message != "" && len(err.Causes) == 0 {
 			errorf := func(path, message string, args ...interface{}) error {
 				contract.Require(path != "", "path")
 				return fmt.Errorf("%s: %s", path, fmt.Sprintf(message, args...))
 			}
 
 			errs = multierror.Append(errs, errorf("#"+err.InstanceLocation, "%v", err.Message))
 		}
-		for _, err := range err.Causes {
-			appendError(err)
+		if len(err.Causes) != 0 {
```

Comment:
Going off of @AaronFriel 's insight https://github.com/pulumi/pulumi/issues/11129, I do think that the last/ innermost error cause takes precedence, such as 
```
        * #/config/bucketNamePrefix: expected string, but got object
        * #/config/bucketNamePrefix: expected integer, but got object
        * #/config/bucketNamePrefix: expected boolean, but got object
        * #/config/bucketNamePrefix: expected array, but got object
        * #/config/bucketNamePrefix/type: value must be one of "string", "integer", "boolean", "array"
```
The first four errors are due to the last error.